Modules/Core/Common/include/itkByteSwapper.hxx

```diff
@@ -63,31 +63,16 @@ ByteSwapper<T>::SwapRangeFromSystemToBigEndian([[maybe_unused]] T * p, [[maybe_u
 
 template <typename T>
 void
-ByteSwapper<T>::SwapWriteRangeFromSystemToBigEndian(const T * p, int num, OStreamType * fp)
+ByteSwapper<T>::SwapWriteRangeFromSystemToBigEndian(const T * p, int num, std::ostream * fp)
 {
-  if constexpr (m_SystemIsBigEndian)
+  if constexpr (m_SystemIsBigEndian || sizeof(T) == 1)
   {
     num *= sizeof(T);
     fp->write(reinterpret_cast<const char *>(p), num);
   }
   else
   {
-    switch (sizeof(T))
-    {
-      case 1:
-        return;
-      case 2:
-        Self::SwapWrite2Range(p, num, fp);
-        return;
-      case 4:
-        Self::SwapWrite4Range(p, num, fp);
-        return;
-      case 8:
-        Self::SwapWrite8Range(p, num, fp);
-        return;
-      default:
-        itkGenericExceptionMacro("Cannot swap number of bytes requested");
-    }
+    SwapWriteRange(p, num, *fp);
```

Comment:
- Just like pull request #4855, this code change is again inspired by the suggestion from Bradley (@blowekamp) at https://github.com/InsightSoftwareConsortium/ITK/pull/4841/files#r1751777344